tests/run/i3539.scala

```diff
@@ -1,7 +1,7 @@
 object Test {
   def main(args: Array[String]): Unit = {
     val i2s = (x: Int) => ""
-    assert(asInstanceOf.asInstanceOf[AnyRef => String].apply(null) == "")
+    assert(i2s.asInstanceOf.asInstanceOf[AnyRef => String].apply(null) == "")
```

Comment:
But the original compilation failure was `(asInstanceOf.asInstanceOf[AnyRef => String].apply(null) == "")` — testing results shouldn't mean dropping regressions tests.

Reply:
That test would clearly fail at runtime. But the same test is bellow.